src/core/undo/MissingPdfUndoAction.cpp

```diff
@@ -0,0 +1,62 @@
+#include "MissingPdfUndoAction.h"
+
+#include "control/Control.h"                         // for Control
+#include "control/PageBackgroundChangeController.h"  // for PageBackGroundChangeController
+#include "gui/MainWindow.h"                          // for MainWindow
+#include "gui/XournalView.h"                         // for XournalView
+#include "model/Document.h"                          // for Document
+#include "model/PageType.h"                          // for PageTypeFormat
+#include "model/XojPage.h"                           // for XojPage
+#include "undo/UndoAction.h"                         // for UndoAction
+#include "util/i18n.h"                               // for _
+
+MissingPdfUndoAction::MissingPdfUndoAction(const fs::path& oldFilepath):
+        UndoAction("MissingPdfUndoAction"), filepath(oldFilepath) {}
+
+MissingPdfUndoAction::~MissingPdfUndoAction() = default;
+
+auto MissingPdfUndoAction::undo(Control* control) -> bool {
+    Document* doc = control->getDocument();
+
+    fs::path redoFilepath = doc->getPdfFilepath();
+
+    doc->resetPdf();
+    doc->setPdfAttributes(filepath, doc->isAttachPdf());
```

Comment:
May it happen, that the value of `doc->isAttachPdf()` may change between those two states?

Reply:
You're right, the user may have changed it when he chose a new PDF background. I will amend the last commit to account for that.